src/main/java/com/minecolonies/entity/ai/citizen/builder/EntityAIStructureBuilder.java

```diff
@@ -344,35 +380,64 @@ private BlockPos getPositionInDirection(EnumFacing facing, int distance)
     @NotNull
     private BlockPos getFloor(@NotNull BlockPos position)
     {
+        final BlockPos floor = getFloor(position, 0);
+        if (floor == null)
+        {
+            return position;
+        }
+        return floor;
+    }
+
+    /**
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) First sentence should end with a period. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=checkstyle%3AJavadoc_Style)